iothub/device/src/WritablePropertyResponse.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class WritablePropertyResponse
+    {
+        private readonly PropertyConvention _propertyConvention;
+
+        /// <summary>
+        /// Convenience constructor for specifying only the property value.
+        /// </summary>
+        /// <param name="propertyValue">The unserialized property value.</param>
+        /// <param name="propertyConvention"></param>
+        public WritablePropertyResponse(object propertyValue, PropertyConvention propertyConvention)
```

Comment:
the writable property response contains user defined payload "Value", so we allow the serialization to be customizable.